### PR TITLE
dlgSpells

### LINKED ISSUE #2772: Simple dry spells dialogue
I suggest a simple dialogue for now - as sort of "partner" to your start dialogue.  I suggest that both (and water balance!) will change, but these will be useful.

1. One missing aspect (to ignore for now) is that there is no mention of "Station" in the existing start dialogue.  For now I suggest similarly for the spells one.  Need to think about this later.  

2. Also, in the start, I don't quite understand the need for the Year and day of year fields?

3. For the start, I assume the resulting worksheet is linked, so another definition would go into the same sheet?  Hence a similar calculation of spell lengths would also (automatically?) go into the same sheet?

4. So, in the new dialogue, instead of threshold you have "Condition".  This has 2 parts to it, with the first having just 3 options, namely "<", "Between" and ">".  Default is "Between" and this then has 2 text boxes open with default values 0 and 0.85.  If < the second box is hidden, and if > the first box is hidden.

5. The "From" and "To" fields remain as you have them.  perhaps they could instead say "From Day:" and "To Day:" - on both dialogues.

6.  So, for now, the spells dialogue will produce just a single column - as does the start of the rains.

7.  There is a checkbox underneath - default unchecked, with label: "Conditional on Rain at start of spell".  If checked this starts the count of the spell length only as 1 on the first day.

That's all for now!

### PR DESCRIPTION
**Pull request by shadrackkibet**
@africanmathsinitiative/developers  this is ready for review. Fixes #2772.

### CODE REVIEW

**Comment by lilyclements** [comment]
Reviewing this now

**Comment by maxwellfundi** [comment]
@lilyclements  i was just about volunteering you to test this.

File: instat/dlgSpells.vb [hunk 1]
```diff
@@ -48,13 +48,13 @@ Public Class dlgSpells
...
-        ucrReceiverRainfall.AddIncludedMetadataProperty("Climatic_Type", {Chr(34) & "rain" & Chr(34)})
-        ucrReceiverRainfall.bAutoFill = True
-        ucrReceiverRainfall.strSelectorHeading = "Rain Variables"
+        ucrReceiverElement.SetParameter(New RParameter("rain", 0))
+        ucrReceiverElement.SetParameterIsRFunction()
+        ucrReceiverElement.bWithQuotes = False
+        ucrReceiverElement.Selector = ucrSelectorForSpells
+        'ucrReceiverElement.AddIncludedMetadataProperty("Climatic_Type", {Chr(34) & "rain" & Chr(34)})
+        ucrReceiverElement.bAutoFill = True
+        ucrReceiverElement.strSelectorHeading = "Variables"
```
Comment (lilyclements) [inline instat/dlgSpells.vb]: This needs to accept only numeric variables.

**Comment by shadrackkibet** [comment]
@rdstern can you clarify this.

**Comment by lilyclements** [comment]
It will need to be a numeric as we are looking at values in a range and can't do that for a factor.
Presumably it can be an integer as well.

**Comment by shadrackkibet** [comment]
what about the issue of Nuds being changed to receivers? I didn't see this at #2772 latest issues to address.

**Comment by shadrackkibet** [comment]
@lilyclements  any other issues on this? apart from the issue of changing From and To Nuds to Receivers which is not yet clear?

**Comment by lilyclements** [comment]
Nope that's it I believe. It looks good other than that

**Comment by lilyclements** [comment]
In light of #3740 can you change the column name here from "Spells" to "spells"?

**Comment by maxwellfundi** [comment]
@shadrackkibet  How is it going with this?

**Comment by shadrackkibet** [comment]
@lilyclements  what can you say about this? is this ready for @dannyparsons ?

File: instat/dlgSpells.vb [hunk 2]
```diff
@@ -275,12 +276,12 @@ Public Class dlgSpells
     End Sub
 
     Private Sub ucrBase_BeforeClickOk(sender As Object, e As EventArgs) Handles ucrBase.BeforeClickOk
-        frmMain.clsRLink.RunScript(clsAddKey.ToScript, strComment:="Spells: Defining column(s) as key")
+        frmMain.clsRLink.RunScript(clsAddKey.ToScript, strComment:="spells: Defining column(s) as key")
```
Comment (lilyclements) [inline instat/dlgSpells.vb]: Can this spells still be a capital? This is the start of a sentence rather than the column name

**Comment by shadrackkibet** [comment]
Sorted.

**Comment by lilyclements** [comment]
excellent, @dannyparsons this is ready to review

**Comment by dannyparsons** [comment]
Why are there changes to sdgCanonicalCorrelation designer? Is this going to cause conflicts with another pull request?

**Comment by shadrackkibet** [comment]
I did  not intend to do this designer edits, I have therefore reverted that commit.

**Comment by dannyparsons** [comment]
Checking all the file changes on Visual Studio before committing will make sure you don't make changes to files accidentally.

**Pull request merged**